### PR TITLE
Only push changes in the output directory

### PR DESCRIPTION
**Pull request by jatofg**
This action runs into problems when there are untracked files present in the repository (in my case, for example, I generate some files and also the Doxyfile dynamically using CMake). Files outside of Doxygen's output directory are then committed and pushed, and when the action runs for a second time then, it just deletes all the files.

I have fixed the script so that it will also work for me, and it seems to work now (you can have a look at the actions in the jatofg/nawa repository if you wish). Feel free to test the changes and merge them.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -22,6 +23,18 @@ GetCurrentBranch () {
...
+PrepareGitHubPagesDirectory() {
+	DESTINATIONDIR=$1
+	
+    # Remove all the files in GitHub Pages directory (if the directory exists)
+	if [ -d "$DESTINATIONDIR" ]; then
+        git rm -rf "$DESTINATIONDIR"
+    fi
+    
+    # Create the GitHub Pages directory if it does not exist
+    mkdir -p "$DESTINATIONDIR"
```
Comment (langroodi) [inline entrypoint.sh:35]: I guess it is better to revert this change to the previous if-else-fi manner. In the current manner the `mkdir` will be execute in any case and `-p` is now saving it from raising an error if the directory exist.
If somebody at the first glance overlooks the `-p` argument or does not know about it, the code block becomes confusing. However, according to me, the previous manner was more readable.

**Comment by langroodi** [comment]
Hi @jatofg! Thanks for your effort. The changes look fine to me. I only had one comment and when it is fixed, I will release the v1.5 version ;)

File: entrypoint.sh (hunk 1 above)
Comment (jatofg) [reply entrypoint.sh:35]: Sorry for the late response, I was quite busy in the last few days. I've added the `-p` argument because otherwise, the directory would not be created recursively (and mkdir would fail if parent directories did not exist). The previous if-else-fi statement was wrong, because `git rm -rf` will not only remove the files in the directory, but also the directory itself. After that, it has to be recreated.

Comment (langroodi) [reply entrypoint.sh:35]: No need to be sorry :) Thanks for your clear explanation. I got your point now.
